src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2092,17 +2104,33 @@ public override void AfterTransactionBegin(ITransaction tx)
 			using (new SessionIdLoggingContext(SessionId))
 			{
 				CheckAndUpdateSessionStatus();
-				Interceptor.AfterTransactionBegin(tx);
+
+				if (!_transactionCoordinatorShared || Interceptor != ConnectionManager.Session.Interceptor)
```

Comment:
I do call it for "children", excepted if that is the same interceptor than the originating session. It does not make sens to call it for each children session in such case, since the originating session will anyway call it, and nothing allow the interceptor to understand why it has many "begin tran" called: it does not know which session has called that.

Previous code was never calling for "children" session, either because the session event was not call at all, or because a test on `_transactionCoordinatorShared` was already there). I have adjusted that for always calling dependent session transaction events and also calling the interceptor only if the dependent session has another interceptor than the originating session.

"Child session" not sharing the connection will have independent transaction handling and will call their interceptor in all cases. Their are just usual sessions indeed, without any knowledge of having been spawned from another session.